sky/clouds/aws.py

```diff
@@ -100,6 +100,21 @@ class AWSIdentityType(enum.Enum):
     #     region                us-east-1      config-file    ~/.aws/config
     SHARED_CREDENTIALS_FILE = 'shared-credentials-file'
 
+    def can_credential_expire(self) -> bool:
+        """Check if the AWS identity type can expire.
+
+        SSO,IAM Role,and Container Role use temporary credentials, which
+        expire and require re-authentication or re-acquisition of credentials.
+        ENV and Shared Credentials File store long-term credentials, which do
+        not expire unless manually changed, but if temporary credentials are
+        used,they will expire and require re-authentication.
+        """
+        expirable_types = {
+            AWSIdentityType.SSO, AWSIdentityType.IAM_ROLE,
+            AWSIdentityType.CONTAINER_ROLE
+        }
```

Comment:
Are we sure these credential types are expiring? We may want to use a CLI command to check expiry of credentials. e.g., claude tells me to use ` sts get-session-token --query 'Credentials.Expiration'`, but we should double check if these actually work.

Reply:
Same for gcloud CLI.

Reply:
In addition, we may want to trigger the CLI check only if IdentityType is one of `AWSIdentityType.SSO, AWSIdentityType.IAM_ROLE, AWSIdentityType.CONTAINER_ROLE`. We may also want to use it with `functools.lru_cache()`. 

This is because CLI check can be expensive and we don't want to slow down `sky jobs launch`.

Reply:
claude result is not correct
```
~/skypilot on dev/hong/controller wip > aws sts get-session-token --query 'Credentials.Expiration'           py sky at 16:28:58

An error occurred (AccessDenied) when calling the GetSessionToken operation: Cannot call GetSessionToken with session credentials
```
we can find session infos from `~/.aws/sso/cache/*json`
Each sso login will generate two json files, one is `aws_sso_credentials.json` the other is `sso_cache.json`
in any of the file we can find `"expiresAt": "2024-12-19T08:22:34Z",`

Reply:
About get-session-token refer to https://docs.aws.amazon.com/cli/latest/reference/sts/get-session-token.html: 
Returns a set of temporary credentials for an Amazon Web Services account or IAM user. The credentials consist of an access key ID, a secret access key, and a security token. Typically, you use GetSessionToken if you want to use MFA to protect programmatic calls to specific Amazon Web Services API operations like Amazon EC2 StopInstances .

MFA-enabled IAM users must call GetSessionToken and submit an MFA code that is associated with their MFA device.